### ISSUE TITLE
Avoid extra git operation to look up "latest" template version

### ISSUE THREAD

**Issue by drevell** [abcxyz/abc#324]
### TL;DR

When installing a template from github with version `@latest`, we do two git operations internally: (1) `git ls-remote` to get the tags and (2) `git clone --depth 1 --branch v1.2.3` to download the template contents. For people like me who use a security key for MFA, this requires two touches of the security key.

A possible alternative: just blindly clone the whole repo, including all branches and tags, rather than trying to optimally clone just the needed tag.

Benefits:
 - Fewer security key touches
 - Probably faster, there's only one connection setup and auth, rather than two.

Downsides:
 - Possibly more downloaded data
